src/server/json_family_test.cc

```diff
@@ -136,9 +136,6 @@ TEST_F(JsonFamilyTest, Type) {
 
   resp = Run({"JSON.TYPE", "json", "$[10]"});
   EXPECT_THAT(resp, ArgType(RespExpr::NIL));
-
-  resp = Run({"JSON.TYPE", "not_exist_key", "$[10]"});
-  EXPECT_THAT(resp, ArgType(RespExpr::NIL_ARRAY));
```

Comment:
what happens now with a non-existent key?

Reply:
It returns an error "no such key", instead of what is written in Elasticache spec to return a NULL array.
I think returning an error is more indicative in this case.

Reply:
It's not up to us to change the spec at this point. I just whitelist these special cases explicitly.
See https://github.com/dragonflydb/dragonfly/blob/main/src/server/zset_family.cc#L1541 for example

Reply:
this still requires a fix :)